### PR TITLE
New brain trauma: expressive aphasia

### PR DESCRIPTION
**Pull request by Lett1**
## About The Pull Request

Adds a new brain trauma: Expressive Aphasia. You can only use the [1000 most common english words](https://splasho.com/upgoer5/phpspellcheck/dictionaries/1000.dicin). Anything not on that list will either get garbled or cause you to forget what you wanted to say mid sentence.

You can try what it's like in [the up-goer 5 editor](https://splasho.com/upgoer5/).

## Why It's Good For The Game

We could always use some more traumas to spice medical up, this is something that the wizard/syndies or admins could unleash on the crew for mass confusion. It's also quite fun to try and work around that limitation.

## Changelog
:cl:
add: New trauma: Expressive Aphasia only. Restricts your speech to the 1000 most common english words.
/:cl:

### CODE REVIEW

**Comment by WJohn** [comment]
Could definitely be a brain damage effect 👍

**Comment by knacker48** [comment]
Feels more like a trauma to me, although I kind of feel mutations and traumas overlap a little too much

**Comment by Lett1** [comment]
> Feels more like a trauma to me, although I kind of feel mutations and traumas overlap a little too much

I could make it into a trait that traumas and mutations can add/remove.

**Comment by Shapsy** [comment]
Should probably have at least a few common SS13 terms and shorthand added as well like HoP, CMO, Captain, medbay, etc, since even the most braindead assistant should be able to call the Captain a comdom

**Comment by JJRcop** [comment]
@Shapsy The top person of this place is stopping the baby maker water

**Comment by Shapsy** [comment]
Oh no

**Comment by Lett1** [comment]
Changed into a brain trauma instead, this makes more sense.

**Comment by Tlaltecuhtli** [comment]
dont we have enough logs to just run a top 1k words used in spessmen?

**Comment by JJRcop** [comment]
It doesn't look like this properly handles suffixes like "-ing", "-er", "-ist", etc.

**Comment by nicbn** [comment]
Should have an option to disable for non-english speaking servers

**Comment by ArcaneMusic** [comment]
"The Bad Hell men are going to make a baby in Computer Man Home."
I love it.

File: code/datums/brain_damage/mild.dm [hunk 1]
```diff
@@ -182,3 +182,44 @@
 			addtimer(CALLBACK(owner, /mob/.proc/emote, "cough"), 12)
 		owner.emote("cough")
 	..()
+
+/datum/brain_trauma/mild/expressive_aphasia
+	name = "Expressive Aphasia"
+	desc = "Patient is affected by partial loss of speech leading to a reduced vocabulary."
+	scan_desc = "inability to form complex sentences"
+	gain_text = "<span class='warning'>You lose the grasp on complex words.</span>"
```
Comment (ATH1909) [inline code/datums/brain_damage/mild.dm]: replace "the' with "your"

**Comment by Barhandar** [comment]
Does it actually tell the players all words they couldn't get out (i.e. not just _one_ arbitrary word) so they can rephrase faster?

**Comment by anconfuzedrock** [comment]
low iq: making expressive aphasia a trauma
infinite iq: making expressive aphasia a quirk

**Comment by Militaires** [comment]
i literally walked in here thinking that it's a quirk until i re-read the title, shit.

**Comment by JJRcop** [comment]
I have a feeling ccc9ff3b6209a3939d82dbb31d410eceffdc61bf has some changes that are not intended to be here? The docs look sweet but are quite large and seem like they are supposed to be used in another change.

**Comment by Lett1** [comment]
> I have a feeling [ccc9ff3](https://github.com/tgstation/tgstation/commit/ccc9ff3b6209a3939d82dbb31d410eceffdc61bf) has some changes that are not intended to be here? The docs look sweet but are quite large and seem like they are supposed to be used in another change. Actually they look like they were auto generated and aren't supposed to be included at all

You're right. This is why you don't do `git add . ` kids.

**Comment by JJRcop** [comment]
You can use `git add -p .` to interactively add changes.

**Comment by optimumtact** [comment]
why use many word when few will do?

File: code/datums/brain_damage/mild.dm [hunk 2]
```diff
@@ -182,3 +182,44 @@
...
+		for(var/word in message_split)
+			var/suffix = copytext(word,-1)
+
+			// Check if we have a suffix and break it out of the word
+			if(suffix in list(".",",",";","!",":","?"))  
+				word = copytext(word,1,-1)
+			else
+				suffix = ""
+
+			word = html_decode(word)
```
Comment (optimumtact) [inline code/datums/brain_damage/mild.dm]: wat

Comment (Lett1) [reply code/datums/brain_damage/mild.dm]: need to decode that otherwise I get html entities in the output

**Comment by optimumtact** [comment]
please fix conflict so I can merge

File: code/datums/brain_damage/mild.dm [hunk 3]
```diff
@@ -182,3 +182,44 @@
...
+/datum/brain_trauma/mild/expressive_aphasia/on_say(message)
+	if(message)
+		var/list/message_split = splittext(message, " ")
+		var/list/new_message = list()
+
+		for(var/word in message_split)
+			var/suffix = copytext(word,-1)
+
+			// Check if we have a suffix and break it out of the word
+			if(suffix in list(".",",",";","!",":","?"))
```
Comment (JJRcop) [inline code/datums/brain_damage/mild.dm]: This list is hard to read

**Comment by Lett1** [comment]
Oh god git what did you do

**Comment by JJRcop** [comment]
If you've rebased to fix the conflicts, this is normal.

**Pull request merged**